### PR TITLE
Add new design goals page

### PR DESCRIPTION
**Pull request by lucperkins**
This addresses issue #860.

### CODE REVIEW

File: website/content/docs/concepts/design-goals.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+Heron is fully API and data model compatible with [Apache
+Storm](http://storm.apache.org), making it easy for developers to transition
+between systems.
+
+### Back pressure
+
+In a distributed system like Heron, there are no guarantees that all system
+components will execute at the same speed. Heron has built-in [back pressure
+mechanisms]({{< ref "#stream-manager" >}}) to ensure that topologies can
```
Comment (objmagic) [inline website/content/docs/concepts/design-goals.md]: ``({{< ref "#stream-manager" >}})`` fails to work now

**Comment by kramasamy** [comment]
@lucperkins - do you have a site where I can look at this?

**Comment by lucperkins** [comment]
I can only display one branch at a time on my personal fork, but I can update it tomorrow

**Comment by kramasamy** [comment]
@lucperkins - sounds good.

**Comment by kramasamy** [comment]
@lucperkins - any update on this?

**Comment by objmagic** [comment]
Please fix ``({{< ref "#stream-manager" >}})`` before merging this.

**Comment by kramasamy** [comment]
@lucperkins - any update on this?

**Comment by lucperkins** [comment]
Yes, I submitted the requested changes

**Comment by objmagic** [comment]
@kramasamy this should be good to go

**Pull request merged**